explainability/core/counterfactuals/src/main/java/org/kie/kogito/explainability/local/counterfactual/CounterfactualConfigurationFactory.java

```diff
@@ -21,46 +21,74 @@
-        solverConfig.setSolutionClass(CounterfactualSolution.class);
+            solverConfig.withEntityClasses(IntegerEntity.class, DoubleEntity.class, BooleanEntity.class, CategoricalEntity.class);
+            solverConfig.setSolutionClass(CounterfactualSolution.class);
 
-        ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = new ScoreDirectorFactoryConfig();
-        scoreDirectorFactoryConfig.setEasyScoreCalculatorClass(CounterFactualScoreCalculator.class);
-        solverConfig.setScoreDirectorFactoryConfig(scoreDirectorFactoryConfig);
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = new ScoreDirectorFactoryConfig();
+            scoreDirectorFactoryConfig.setEasyScoreCalculatorClass(CounterFactualScoreCalculator.class);
+            solverConfig.setScoreDirectorFactoryConfig(scoreDirectorFactoryConfig);
 
-        TerminationConfig terminationConfig = new TerminationConfig();
-        terminationConfig.setSecondsSpentLimit(timeLimit);
-        solverConfig.setTerminationConfig(terminationConfig);
+            solverConfig.setTerminationConfig(terminationConfig);
 
-        LocalSearchAcceptorConfig acceptorConfig = new LocalSearchAcceptorConfig();
-        acceptorConfig.setEntityTabuSize(tabuSize);
+            LocalSearchAcceptorConfig acceptorConfig = new LocalSearchAcceptorConfig();
+            acceptorConfig.setEntityTabuSize(tabuSize);
 
-        LocalSearchForagerConfig localSearchForagerConfig = new LocalSearchForagerConfig();
-        localSearchForagerConfig.setAcceptedCountLimit(acceptedCount);
+            LocalSearchForagerConfig localSearchForagerConfig = new LocalSearchForagerConfig();
+            localSearchForagerConfig.setAcceptedCountLimit(acceptedCount);
 
-        LocalSearchPhaseConfig localSearchPhaseConfig = new LocalSearchPhaseConfig();
-        localSearchPhaseConfig.setAcceptorConfig(acceptorConfig);
-        localSearchPhaseConfig.setForagerConfig(localSearchForagerConfig);
+            LocalSearchPhaseConfig localSearchPhaseConfig = new LocalSearchPhaseConfig();
+            localSearchPhaseConfig.setAcceptorConfig(acceptorConfig);
+            localSearchPhaseConfig.setForagerConfig(localSearchForagerConfig);
 
-        List<PhaseConfig> phaseConfigs = new ArrayList<>();
-        phaseConfigs.add(localSearchPhaseConfig);
+            List<PhaseConfig> phaseConfigs = new ArrayList<>();
+            phaseConfigs.add(localSearchPhaseConfig);
+
+            solverConfig.setPhaseConfigList(phaseConfigs);
+            return solverConfig;
+        }
+
+        public Builder withTabuSize(int size) {
+            this.tabuSize = size;
+            return this;
+        }
+
+        public Builder withAcceptedCount(int count) {
+            this.acceptedCount = count;
+            return this;
+        }
+
+        public Builder withTerminationConfig(TerminationConfig terminationConfig) {
+            this.terminationConfig = terminationConfig;
+            return this;
+        }
```

Comment:
What about also provide an additional helper method to directly accept `scoreCalculationCountLimit` that seems to be the most common parameter? This should simplify the usage (i.e. in the tests)
```suggestion
        public Builder withTerminationConfig(TerminationConfig terminationConfig) {
            this.terminationConfig = terminationConfig;
            return this;
        }
        
        public Builder withScoreCalculationCountLimit(long scoreCalculationCountLimit) {
            if (this.terminationConfig == null) {
                this.terminationConfig = new TerminationConfig();
            }
            this.terminationConfig.setScoreCalculationCountLimit(10L);
            return this;
        }
```

Reply:
Good point, thanks! Implemented.